File: frontend/src/app/model/config/course.ts
import { AuditAbleModel } from "../super-model/audit-able-model";
import { TaughtCourse } from "./taught-course"; // Import TaughtCourse if needed

export interface Course extends AuditAbleModel {
    id?: number;
    courseCode?: string;
    courseName?: string;
    description?: string;
    courseMaterials?: string;
    syllabus?: string;
    
}

File: backend/src/main/java/com/sajjad/academicportalbackend/model/config/Course.java
package com.sajjad.academicportalbackend.model.config;

import com.sajjad.academicportalbackend.model.super_classes.AuditableEntity;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@Entity
public class Course extends AuditableEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String courseCode;
    private String courseName;
    private String description;
    private String courseMaterials;
    private String syllabus;


    @ManyToMany(fetch = FetchType.EAGER)
    private List<TeacherProfile> teachers;







}
